2-mandatory/3-shopping-list.js

```diff
@@ -20,6 +20,18 @@ let pantry = {
 
 function createShoppingList(recipe) {
   // write code here
+  let shoppingList = {
+    name: recipe.name,
+    items: []
+  };
+
+  for (let ingredient of recipe.ingredients) {
+    if (!pantry.fridgeContents.includes(ingredient) && !pantry.cupboardContents.includes(ingredient)) {
```

Comment:
well done, you can also use concat to merge two arrays and check once if the value is in there or not , so you can merge the   `pantry.fridgeContents` and `pantry.cupboardContents` in one array and check includes for that array, does this makes sense? let me know if you need more explanation about this :)